### PR TITLE
use parent child pom mode instead of submodule

### PR DESCRIPTION
**Pull request by zhexuany**
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tispark/159)

### CODE REVIEW

**Comment by zhexuany** [comment]
@birdstorm @Novemser @ilovesoup PTAL

**Comment by Novemser** [comment]
I suggest we first merge the remaining PRs left in client side and then migrant to parent/child mode.

File: core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+      <groupId>com.pingcap.tispark</groupId>
+      <artifactId>tispark-parent</artifactId>
+      <version>0.1.0-SNAPSHOT</version>
+      <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>tispark-core</artifactId>
+    <version>0.1.0-SNAPSHOT</version>
```
Comment (ilovesoup) [inline core/pom.xml]: why do we need a separated version for modules

File: pom.xml [hunk 2]
```diff
@@ -20,12 +33,13 @@
         <scala.binary.version>2.11</scala.binary.version>
         <scala.version>2.11</scala.version>
     </properties>
+
+    <modules>
+        <module>tikv-client</module>
```
Comment (ilovesoup) [inline pom.xml]: shall we put integration test here as well?

File: core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    <packaging>jar</packaging>
+    <name>TiSpark Project Core</name>
+    <url>http://github.copm/pingcap/tispark</url>
+    <licenses>
+      <license>
```
Comment (ilovesoup) [inline core/pom.xml]: also these I think need to be gone and keep one copy in parent only.
e.g, organizations, license and etc.

File: tikv-client/pom.xml [hunk 4]
```diff
@@ -0,0 +1,337 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.pingcap.tikv</groupId>
+    <artifactId>tikv-client</artifactId>
+    <version>0.1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+    <name>TiSpark Project TiKV Java Client</name>
+    <organization>
```
Comment (ilovesoup) [inline tikv-client/pom.xml]: move these to parent as well.

**Comment by ilovesoup** [comment]
1. Please resolve conflict.
2. Move README from core to root and add instruction of how to build for new structure.
3. Please make sure project is buildable and pass tests.
4. Please address the comments.
5. For other parts, LGTM.

**Comment by birdstorm** [comment]
fix #93

File: core/pom.xml (hunk 1 above)
Comment (zhexuany) [reply core/pom.xml]: check this one. https://github.com/apache/spark/blob/master/core/pom.xml#L24  I am not sure reason but spark do the same way.

File: pom.xml (hunk 2 above)
Comment (zhexuany) [reply pom.xml]: DONE, push commit later.

File: tikv-client/pom.xml (hunk 4 above)
Comment (zhexuany) [reply tikv-client/pom.xml]: DONE

**Comment by zhexuany** [comment]
- [ ] archived tikv-java-client
- [ ] pass all tests
- [ ] check readme and add build method

**Comment by zhexuany** [comment]
DAG Test
Result: Tests run: 6841  Tests succeeded: 3774  Tests failed: 8  Tests skipped: 3059
Index Test
Result: Tests run: 1228  Tests succeeded: 691  Tests failed: 206  Tests skipped: 331

**Comment by zhexuany** [comment]
Result: Tests run: 21  Tests succeeded: 21  Tests failed: 0  Tests skipped: 0

**Pull request merged**